src/Dropdown/Dropdown.jsx

```diff
@@ -171,6 +177,76 @@ const Dropdown = ({
     [disabled, error, isCollapsed]
   )
 
+  useLayoutEffect(() => {
+    const handleKeyDownEvent = e => {
+      if (isCollapsed) {
+        return
+      }
+
+      if (e.code === 'Escape') {
+        e.preventDefault()
+
+        setCollapsed(true)
+      }
+
+      if (e.code === 'ArrowDown') {
+        e.preventDefault()
+
+        setCurrentKeyboardFocus(hover =>
+          Math.min(options.length - 1, Math.max(-1, hover + 1))
+        )
+      }
+
+      if (e.code === 'ArrowUp') {
+        e.preventDefault()
+
+        setCurrentKeyboardFocus(hover =>
+          Math.min(options.length - 1, Math.max(-1, hover - 1))
+        )
+      }
+
+      if (e.code === 'Enter') {
+        e.preventDefault()
+
+        if (
+          currentKeyboardFocus !== -1 &&
+          !options[currentKeyboardFocus].props.disabled
+        ) {
+          onSelectItem(options[currentKeyboardFocus].props.value)
+        }
+      }
+    }
+
+    const removeCurrentHover = () => {
+      setCurrentKeyboardFocus(-1)
+    }
+
+    const element = wrapper.current
+    element.addEventListener('keydown', handleKeyDownEvent)
+    element.addEventListener('mousemove', removeCurrentHover)
+    return () => {
+      element.removeEventListener('keydown', handleKeyDownEvent)
+      element.removeEventListener('mousemove', removeCurrentHover)
+    }
+  }, [options, currentKeyboardFocus, onSelectItem, isCollapsed])
+
+  useEffect(() => {
+    if (isCollapsed) {
+      setCurrentKeyboardFocus(-1)
+      return
+    }
+
+    if (multiple) {
+      return
+    }
+
+    const optionIndex = options.findIndex(option =>
+      isEqual(option.props.value, value)
+    )
+
+    setCurrentKeyboardFocus(optionIndex)

```

Comment:
I think we can skip this `.find` and use `setCurrentKeyboardFocus(optionIndex)` inside of the children(`<Option {...existingProps} index={index} />`) exactly when we are selecting the option

Reply:
This is used when we open an already selected dropdown and we need to set the current option index to the selected one. It does not make sense to move that responsibility in the children, especially when we only need this to run when the dropdown opens